### PR TITLE
Fix for #1137

### PR DESCRIPTION
**Pull request by chrisforbes**
This tweaks the renderpass validation to not fail on out-of-range attachment indices.

There's more to do here -- #1080 etc -- but avoiding this failure case is a good start.

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -2786,6 +2786,26 @@ TEST_F(VkLayerTest, MismatchedQueueFamiliesOnSubmit) {
     m_errorMonitor->VerifyFound();
 }
 
+TEST_F(VkLayerTest, RenderPassAttachmentIndexOutOfRange) {
```
Comment (mark-lunarg) [inline tests/layer_validation_tests.cpp]: clang-format?

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -10258,7 +10258,7 @@ static bool ValidateAttachmentIndex(layer_data *dev_data, uint32_t attachment, u
     if (attachment >= attachment_count && attachment != VK_ATTACHMENT_UNUSED) {
         skip_call |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, (VkDebugReportObjectTypeEXT)0, 0, __LINE__,
                              DRAWSTATE_INVALID_ATTACHMENT_INDEX, "DS",
-                             "CreateRenderPass: %s attachment %d cannot be greater than the total number of attachments %d.",
+                             "CreateRenderPass: %s attachment %d must be less than the total number of attachments %d.",
```
Comment (tobine) [inline layers/core_validation.cpp]: I believe this is VALIDATION_ERROR_00325. Please add enum along with msg and update database.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -2786,6 +2786,26 @@ TEST_F(VkLayerTest, MismatchedQueueFamiliesOnSubmit) {
...
+        { 0, VK_PIPELINE_BIND_POINT_GRAPHICS, 0, nullptr, 1, &ref, nullptr, nullptr, 0, nullptr },
+    };
+
+    VkRenderPassCreateInfo rpci = {
+        VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO, nullptr,
+        0, 0, nullptr, 1, subpasses, 0, nullptr
+    };
+    VkRenderPass rp;
+
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT,
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: Once unique enum is added, can validate this based on enum.

**Comment by chrisforbes** [comment]
Issues fixed, and pushed to master.

**Pull request closed without merging**